lib/supavisor/helpers.ex

```diff
@@ -88,7 +88,11 @@ defmodule Supavisor.Helpers do
   @spec get_user_secret(pid(), String.t(), String.t()) :: {:ok, map()} | {:error, String.t()}
   def get_user_secret(conn, auth_query, user) do
     try do
-      Postgrex.query!(conn, auth_query, [user])
+      # sanitize the user input by removing all characters that are not alphanumeric or underscores
+      user = String.replace(user, ~r/[^a-zA-Z0-9_]/, "")
```

Comment:
@filipecabaco wdyt?

Reply:
I think that instead we should reject `user` if these characters are present instead of just pretending that these do not exists.

Reply:
I already have a check for invalid characters during connection, but it only checks for slashes. I added this sanitization as an additional guard. It might be worth applying the same rule (`a-zA-Z0-9_`) during connection as well

https://github.com/supabase/supavisor/blob/61b629dc1995a6c1245c6d359c7a9f647181421b/lib/supavisor/client_handler.ex#L157-L164

```bash
$ psql postgresql://p\"ostgres.tenant:password@localhost:5432/postgres
psql: error: connection to server at "localhost" (::1), port 5432 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?
connection to server at "localhost" (127.0.0.1), port 5432 failed: FATAL:  Authentication error, reason: "Invalid characters in user or db_name"
```